src/modules/sub-items/components/pagination/pagination.component.js

```diff
@@ -0,0 +1,126 @@
+
+export default class PaginationComponent extends PureComponent {
+    getPagesNumber() {
+        const { totalCount, itemsPerPage } = this.props;
+
+        return Math.ceil(totalCount / itemsPerPage);
+    }
+
+    /**
+     * Computes array with pagination pages.
+     *
+     * Example 1: [ 1, "...", 5, 6, 7, 8, 9, 10 ] (for: proximity = 2; pagesNumber = 10; currentPage = 7)
+     * Example 2: [ 1, "...", 3, 4, 5, 6, 7, "...", 10 ] (for: proximity = 2; pagesNumber = 10; currentPage = 5)
+     * Example 3: [ 1, "...", 8, 9, 10, 11, 12, "...", 20 ] (for: proximity = 2; pagesNumber = 20; currentPage = 10)
+     *
+     * @returns {Array}
+     */
+    computePages() {
+        const { proximity, currentPage } = this.props;
+        const pagesNumber = this.getPagesNumber();
+        const pages = [];
+        let wereDots = false;
+
+        for (let i = 1; i <= pagesNumber; i++) {
+            const isFirstPage = i === 1;
+            const isLastPage = i === pagesNumber;
+            const isInRange = i >= (currentPage + 1) - proximity && i <= (currentPage + 1) + proximity;
+
+            if (isFirstPage || isLastPage || isInRange) {
+                pages.push(i);
+                wereDots = false;
+            } else if (!wereDots) {
+                pages.push('...');
+                wereDots = true;
+            }
+        }
+
+        return pages;
+    }
+
+    renderPrevious() {
+        const { currentPage, onPageChange } = this.props;
+        const previousLabel = Translator.trans(/*@Desc("Back")*/ 'pagination.previous', {}, 'sub_items');
+        const previousPage = currentPage - 1;
+        const isFirstPage = currentPage === 0;
+
+        return (
+            <PaginationBtnComponent
+                page={previousPage}
+                label={previousLabel}
+                additionalClasses="prev"
+                disabled={isFirstPage}
+                onPageChange={onPageChange}
+            />
+        );
+    }
+
+    renderNext() {
+        const { currentPage, onPageChange } = this.props;
+        const nextLabel = Translator.trans(/*@Desc("Next")*/ 'pagination.next', {}, 'sub_items');
+        const nextPage = currentPage + 1;
+        const isLastPage = currentPage === this.getPagesNumber() - 1;
+
+        return (
+            <PaginationBtnComponent
+                page={nextPage}
+                label={nextLabel}
+                additionalClasses="next"
+                disabled={isLastPage}
+                onPageChange={onPageChange}
+            />
+        );
+    }
+
+    renderPages(pages) {
+        const { currentPage, onPageChange } = this.props;
+
+        return pages.map((page, index) => {
+            if (page === '...') {
+                return <PaginationBtnComponent key={index} label="..." disabled={true} />;
+            }
+
+            const isCurrentPage = page === currentPage + 1;
+            const additionalClasses = isCurrentPage ? 'active' : '';
```

Comment:
Is `active` CSS class coming from Bootstrap?

Reply:
Yes, https://getbootstrap.com/docs/4.1/components/pagination/#disabled-and-active-states